### PR TITLE
PRS-3213: Jenkins pipeline config

### PR DESCRIPTION
**Pull request by meyerhp**
### Description
Jenkins pipeline config to replace TravisCI

Pipeline syntax documentation: https://www.jenkins.io/doc/book/pipeline/syntax/

### Testing
Code review is fine for now. The plan is to deploy a Jenkins server either later this week or next week so we can test it end to end.

### Issue
https://spandigital.atlassian.net/browse/PRSDM-3212

### Checklist before merging

* [x] Is this code covered by tests?
* [x] Is the documentation updated for this change?

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// To run this Jenkinsfile, you need to have the following plugins installed:
+// - NodeJS
+// - CloudBees AWS Credentials
+// - Pipeline: AWS Steps
+pipeline {
+    agent any
+    tools {nodejs "19.4"}
```
Comment (effleurager) [inline Jenkinsfile:7]: Why not stick with an LTS?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            }
+        }
+        stage("Github Release") {
+            when{
+                buildingTag()
+            }
+            steps {
+                script {
+                    sh "curl -v -d \"{\"tag_name\": \"$TAG_NAME\"}\" -H \"Authorization: token $github_key\" -X POST \"https://api.github.com/repos/presidium-js/releases\""
```
Comment (effleurager) [inline Jenkinsfile:82]: Specifying `POST` is redundant when using `-d`
```suggestion
                    sh "curl -v -d \"{\"tag_name\": \"$TAG_NAME\"}\" -H \"Authorization: token $github_key\" https://api.github.com/repos/presidium-js/releases"
```

Comment (effleurager) [reply Jenkinsfile:82]: Thoughts on replacing raw request with a proper tool? https://cli.github.com/manual/gh_release_create

**Review by Zalaras — APPROVED** [review]
LGTM

File: Jenkinsfile (hunk 1 above)
Comment (meyerhp) [reply Jenkinsfile:7]: This was mostly for testing. We can switch out the version when we set up the Jenkins server

**Pull request merged**